cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -157,45 +160,46 @@ func generateMountsModeRun() error {
 	return fmt.Errorf("run mode mount generation not implemented yet")
 }
 
-func isInstallMode(content []byte) bool {
-	// XXX: deal with whitespace
-	if bytes.Contains(content, []byte("snapd_recovery_mode=install")) {
-		return true
+var validModes = []string{"install", "recover", "run"}
+
+func whichMode(content []byte) (string, error) {
+	scanner := bufio.NewScanner(bytes.NewBuffer(content))
```

Comment:
Maybe we could have a cmdline parser helper? It's possible that we'll need to parse this again elsewhere in the code.